src/html/HTMLWriter.jl

```diff
@@ -745,6 +745,79 @@ const _SEARCHRECORD_IGNORED_BLOCK_TYPES = Union{
     Documenter.DocsNodesBlock,
     Documenter.SetupNode,
 }
+
+# Content segment for grouping content by sections
+struct ContentSegment
+    section_header::Union{Node, Nothing}  # AnchoredHeader node or nothing for default section
+    content_nodes::Vector{Node}           # Content nodes belonging to this section
+end
+
+# Segment page content by sections for improved search indexing
+function segment_page_by_sections(page_mdast::Node)
+    segments = ContentSegment[]
+    current_segment_content = Node[]
+    current_section_header = nothing
+
+    for node in page_mdast.children
+        if node.element isa Documenter.AnchoredHeader
+            # Save previous segment if it has content or a header
+            if !isempty(current_segment_content) || current_section_header !== nothing
+                push!(segments, ContentSegment(current_section_header, copy(current_segment_content)))
+            end
+            # Start new section
+            current_section_header = node
+            current_segment_content = Node[]
+        else
+            # Skip nodes that shouldn't be indexed
+            if !(node.element isa _SEARCHRECORD_IGNORED_BLOCK_TYPES)
+                push!(current_segment_content, node)
+            end
+        end
+    end
+
+    # Add final segment
+    if !isempty(current_segment_content) || current_section_header !== nothing
+        push!(segments, ContentSegment(current_section_header, current_segment_content))
+    end
+
+    return segments
+end
+
+# Create search record for a content segment
+function create_segment_search_record(ctx::HTMLContext, navnode::Documenter.NavNode, segment::ContentSegment)
+    if segment.section_header === nothing
+        # Default section - use page title and aggregate content
+        page_title = mdflatten_pagetitle(DCtx(ctx, navnode))
+        content_text = isempty(segment.content_nodes) ? "" :
+            join([mdflatten(node) for node in segment.content_nodes], "\n\n")
```

Comment:
Isn't this just the same as
```suggestion
        content_text = join([mdflatten(node) for node in segment.content_nodes], "\n\n")
```